python/graphstorm/model/gnn_distill/student.py

```diff
@@ -0,0 +1,144 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    GNN student model class
+"""
+import os
+import torch as th
+from torch import nn
+from transformers import (
+    AutoTokenizer,
+    AutoConfig,
+    AutoModel,
+    DistilBertConfig,
+    DistilBertModel,
+)
+
+SUPPORTED_MODEL = {
+"DistilBertModel": (DistilBertConfig, DistilBertModel),
+}
+
+PRETRAINED_MODEL = [
+"distilbert-base-uncased",
+"distilbert-base-cased",
+]
+
+class GSDistilledModel(nn.Module):
+    """ GraphStorm GNN distilled model class.
+    User specified Transformer-based model 
+    and a projection matrix will be initialized.
+
+    Parameters
+    ----------
+    lm_type : str
+        Model type for Transformer-based student model.
+    pre_trained_name : str
+        Name of pre-trained model.
+    checkpoint_path : str
+        Path of model checkpoint.
+    """
+    def __init__(self, lm_type=None, pre_trained_name=None, checkpoint_path=None):
+        super(GSDistilledModel, self).__init__()
+
+        if not checkpoint_path:
```

Comment:
Let's add a TODO here. Basically, we can unify the implementation of between L55 to L76.
Both use AutoTokenizer.from_pretrained and lm.from_pretrained

Reply:
added